### PR TITLE
feat(lotdetails): L3-3145 pinch zoom component in seldon

### PR DESCRIPTION
**Pull request by sgrund14**
**Jira ticket**

[L3-3145](https://phillipsauctions.atlassian.net/browse/L3-3145)

**Screenshots**

https://github.com/user-attachments/assets/76198f78-376d-4d8b-8fa7-02b4ee50af48

**Summary**

A pinch zoom component in seldon
Nicely accompanied by the carousel component to form a pinch zoom carousel

To be accompanied by a remix PR shortly

**Change List (describe the changes made to the files)**

- New Pinch Zoom Component
- Stories demonstrating pinch zoom component when combined with both carousel and modal
- Tiny test suite (need better way to simulate wheel in / out, but otherwise not too much to test?)
- Small updates to carousel & modal components to allow more customization

**Acceptance Test (how to verify the PR)**

- Add step by step instructions on how to verify the change

**Regression Test**

- (Optional) Add verification steps to make sure this PR doesn't break old functionality

**Evidence of testing**

- Post logs, screenshots, etc

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-3145]: https://phillipsauctions.atlassian.net/browse/L3-3145?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
     "flatpickr": "^4.6.13",
     "html-react-parser": "^5.1.12",
     "react-modal": "^3.16.1",
+    "react-quick-pinch-zoom": "^5.1.0",
```
Comment (scottdickerson) [inline package.json:57]: this isn't necessarily an indictment of this package, but I do see a much more widely used react zoom package according to npmtrends: 
image
Did you have a chance to check that one out?

File: src/components/Carousel/_carousel.scss [hunk 2]
```diff
@@ -21,6 +21,10 @@
     flex: 0 0 100%;
     min-width: 0;
 
+    &--cursor-pointer {
```
Comment (scottdickerson) [inline src/components/Carousel/_carousel.scss:24]: this is such a tailwindy thing :)

Comment (scottdickerson) [reply src/components/Carousel/_carousel.scss:24]: could put in the `utils.scss` as a mixin to make it more scss-y

File: src/components/PinchZoom/PinchZoom.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      containerProps={{
+        ...commonProps,
+        ...props,
+        className: classNames(baseClassName, className, {
+          [`${baseClassName}-zoomed`]: isZoomed,
+        }),
+      }}
+    >
+      {children ? (
+        cloneElement(children, {
```
Comment (scottdickerson) [inline src/components/PinchZoom/PinchZoom.tsx:78]: might be good to check `isValidElement` before cloning.

File: vite.config.ts [hunk 4]
```diff
@@ -22,6 +22,7 @@ export default defineConfig({
     target: ['es2019'],
     minify: true,
     reportCompressedSize: true,
+    commonjsOptions: { transformMixedEsModules: true },
```
Comment (scottdickerson) [inline vite.config.ts:25]: yikes, maybe another reason to check the other library out.  I think this is kind of an indicator that the library is a bit screwy.

File: src/components/Carousel/CarouselDots.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import classNames from 'classnames';
 import { EmblaCarouselType } from 'embla-carousel';
-import { ComponentProps, forwardRef, useCallback, useEffect, useState } from 'react';
+import { ComponentProps, forwardRef, useCallback, useEffect, useState, useId } from 'react';
```
Comment (davidicus) [inline src/components/Carousel/CarouselDots.tsx:3]: 🤯 I did not know React exported something like this. Very cool

**Review by davidicus — APPROVED** [review]

**Review by scottdickerson — CHANGES_REQUESTED** [review]
This is working well, but I wonder if we should trigger zoom-in and zoom-out on a single click and use the `zoom-in` cursor when not zoomed in

**Review by scottdickerson — APPROVED** [review]
LGTM

**Pull request merged**